websocket_handler.c

```diff
@@ -24,9 +24,13 @@ void websocket_event_handler(void *handler_args, esp_event_base_t base, int32_t
             break;
 
         case WEBSOCKET_EVENT_DATA:
-            ESP_LOGI(TAG, "WEBSOCKET_EVENT_DATA");
-            ESP_LOGI(TAG, "Received opcode=%d", data->op_code);
-            ESP_LOGW(TAG, "Received=%.*s\r\n", data->data_len, (char*)data->data_ptr);
+        	//Avoid printing logs if it is just a ping/pong frame.
+        	if(websocket_op_ping_frame != data->op_code && websocket_op_pong_frame != data->op_code)
```

Comment:
```suggestion
        	if(data->op_code  != websocket_op_ping_frame || data->op_code != websocket_op_pong_frame)
```

I think it should be or, since and will never become true, as a variable can't have two values at once.

Reply:
Actually no. If it is OR instead of AND then 

Let's say when ping came in, one of the conditions is true and it will print for that ping which is not wanted. And same for pong frame. 

Also, better practice is (pong_enum != Opcode) than otherwise because it prevents unwanted value assignment to variable. Let's say if you forget to put **!** then value will be assigned to opcode and statement will become true.

Reply:
> Actually no. If it is OR instead of AND then
> 
> Let's say when ping came in, one of the conditions is true and it will print for that ping which is not wanted. And same for pong frame.

Ah, right !! my bad.

> Also, better practice is (pong_enum != Opcode) than otherwise because it prevents unwanted value assignment to variable. Let's say if you forget to put **!** then value will be assigned to opcode and statement will become true.

umm, okay. Never used enums, didn't know about this, but in case it was a normal variable I would be right ?

Reply:
If you are comparing variable with a macro or a constant it is still a better idea to keep the variable on right because at compilation, compiler will complain about the wrong syntax if you missed **!**.

If you are comparing 2 variables, any sequence is fine but they are prone to hardware hack in a system where you didn't want anyone to read data or change while accessing. Let's say if the variable is passed by reference, if you change the value by mistake you might alter the actual data structure which is unacceptable. In that case you can receive the variable in function as a const, so even when you passed variable to a function by reference, the local function is receiving it as a constant and will be unchangeable for that scope and also safe from hacking(reverse engineering).

Although, I personally use any sequence until I am comparing it with a constant or enum.

Reply:
TIL, I never gave this a thought, but thanks for letting me know :) Need more experience with development :D

Reply:
True, embedded is huge. Even I am learning. I learned that while watching how to hack a firmware or binary file and loopholes in C language. There are tools in linux which actually helps you to get file structure of linux based code from hex or bin file. Anyway I'll update the enum and request again.

Reply:
Done.

Reply:
Thanks for your contribution.

Reply:
> There are tools in linux which actually helps you to get file structure of linux based code from hex or bin file.

This is interesting